src/Container.php

```diff
@@ -166,6 +166,11 @@ public function get($id, bool $new = false)
 
         if ($this->providers->provides($id)) {
             $this->providers->register($id);
+            
+            if(!$this->definitions->has($id) && !$this->definitions->hasTag($id)) {
+                throw new ContainerException(sprintf('Service provider lied about providing (%s) service', $id));    
```

Comment:
its actually possible to get all service providers and check which one returned true when `provides` is called to blame, not sure whether i should add that or no here, in nuxed, `$this->providers->provides($id)` returns `(bool, ?string)`, `(true, "service-provider-id")` or `(false, null)` but php doesn't have `tuple` type and it would be a BC break. another option is to add a method to the `ServerProviderAggregateInterface` that tells which service provider claimed to provide that specific service, but again that's a BC break.

cc @philsturgeon

Reply:
Only just seen this, you tagged the wrong Phil, I agree that it should be an exception as it's not something that should ever make it to production, hence why it never threw before, allowing it to bubble out was fine for me, but an exception would be more clear, not a fan of the exception message though, so I'll update that before I merge in the next few days